src/LibraryManager/Providers/BaseProvider.cs

```diff
@@ -0,0 +1,297 @@
+                try
+                {
+                    foreach (string file in state.Files)
+                    {
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            return LibraryOperationResult.FromCancelled(state);
+                        }
+
+                        if (string.IsNullOrEmpty(file))
+                        {
+                            return new LibraryOperationResult(state, PredefinedErrors.CouldNotWriteFile(file));
+                        }
+
+                        string destinationPath = Path.Combine(state.DestinationPath, file);
+                        var sourceStream = new Func<Stream>(() => GetStreamAsync(state, file, cancellationToken).Result);
+                        bool writeOk = await HostInteraction.WriteFileAsync(destinationPath, sourceStream, state, cancellationToken).ConfigureAwait(false);
+
+                        if (!writeOk)
+                        {
+                            return new LibraryOperationResult(state, PredefinedErrors.CouldNotWriteFile(file));
+                        }
+                    }
+                }
+                catch (UnauthorizedAccessException)
+                {
+                    return new LibraryOperationResult(state, PredefinedErrors.PathOutsideWorkingDirectory());
+                }
+                catch (Exception ex)
+                {
+                    HostInteraction.Logger.Log(ex.ToString(), LogLevel.Error);
+                    return new LibraryOperationResult(state, PredefinedErrors.UnknownException());
+                }
+            }
+
+            return LibraryOperationResult.FromSuccess(state);
+        }
+
+        private async Task<Stream> GetStreamAsync(ILibraryInstallationState state, string sourceFile, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrEmpty(state.Name) && !string.IsNullOrEmpty(state.Version))
+            {
+                string absolute = Path.Combine(CacheFolder, state.Name, state.Version, sourceFile);
+
+                if (File.Exists(absolute))
+                {
+                    return await HostInteraction.ReadFileAsync(absolute, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            return null;
+        }
+
+        private bool IsLibraryUpToDate(ILibraryInstallationState state)
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(state.Name) && !string.IsNullOrEmpty(state.Version))
+                {
+                    string cacheDir = Path.Combine(CacheFolder, state.Name, state.Version);
+                    string destinationDir = Path.Combine(HostInteraction.WorkingDirectory, state.DestinationPath);
+
+                    foreach (string sourceFile in state.Files)
+                    {
+                        var destinationFile = new FileInfo(Path.Combine(destinationDir, sourceFile).Replace('\\', '/'));
```

Comment:
This is normalizing to use '/', which is the only path separator we should be using.  (We don't even allow `\` in the `"destination"` value through the manifest schema, and I'm surprised that we do allow it on the `"files"` items.  Probably an oversight there, since it allows a mix.)